javascript/javascript1/week4/Voice-assistant.js

```diff
@@ -0,0 +1,150 @@
+let name = '';
+const todo = [];
+
+function getReply(command){
+  command = command.toLowerCase();
+  if(command.startsWith('hello my name is')){
+  return greeting(command)
+  }
+
+  if(command === 'what is my name?'){
+   return whatIsName() 
+  }
+   
+  if(command.startsWith('add') && command.includes('to my todo')){
+    return addTodo(command);
+  }
+
+
+  if(command.startsWith('remove') && command.includes('from my todo')){
+    const task = command.split(' ').slice(1, command.split(' ').length - 3).join(' ');
+    removeItem(todo, task);
+  }
+
+
+  if(command === 'what is on my todo?'){
+    let items = writeItems(todo)
+    return `You have ${todo.length} todos -${items}`
+  }
+
+
+  if(command==='what day is it today?'){
+    return findToday();
+  }
+    
+  if (command.startsWith('what is')){
+    return simpleMath(command)
+  }
+
+  if(command.startsWith('set a timer for'))
+    return timer(command);
+
+  if(command === 'who created you?')
+    return `Farzaneh Ahmadi!:)`
+}
+
+
+function greeting(command){
+  if(name === ''){
+    name =command.split(' ').slice(4).join(' ');
+    name = capitalizedFirstLetter(name)
+    return `Nice to meet you ${name}`}
+  else if(name !== '' && name === capitalizedFirstLetter(command.split(' ').slice(4).join(' ')) )
+    return `Hi agian, ${name}! Nice to meet you!`
+  else if(name !== '' && name!==command.split(' ').slice(4).join(' ')){
+    name=command.split(' ').slice(4).join(' ');
+    name = capitalizedFirstLetter(name)
+    return `Nice to meet you ${name}`}
+}
+
+function capitalizedFirstLetter(str){
+  const firstLetter = str.charAt(0);
+  const firstLetterCap = firstLetter.toUpperCase()
+  const remainingLetters = str.slice(1);
+  return firstLetterCap + remainingLetters
+}
+
+function whatIsName(){
+  if(name !== '')
+    return `${name}`
+  else
+    return `I don't know your name yet!`
+}
+
+
+function addTodo(command){
```

Comment:
It would be good to return a message with what item was added to todo list, like "Fishing added to your todo list"